### ISSUE TITLE
About learning rate scheduler

### ISSUE THREAD

**Issue by Vincent-Hoo** [pytorch/pytorch#31871]
here is my code
```
optimizer = optim.SGD(net.parameters(), lr=0.1)
scheduler = optim.lr_scheduler.StepLR(optimizer, step_size=5, gamma=0.5)

for i in range(15):
    lr = scheduler.get_lr()[0]
    lr1 = optimizer.param_groups[0]["lr"]
    print(i, lr, lr1)
    scheduler.step()
```

And here is the result
```
0 0.1 0.1
1 0.1 0.1
2 0.1 0.1
3 0.1 0.1
4 0.1 0.1
5 0.025 0.05
6 0.05 0.05
7 0.05 0.05
8 0.05 0.05
9 0.05 0.05
10 0.0125 0.025
11 0.025 0.025
12 0.025 0.025
13 0.025 0.025
14 0.025 0.025
```

We can see that the when `scheduler.step()` is applied, the learning rate first decreases 0.25 times, then bounces back to 0.5 times. Is it the problem of `scheduler.get_lr()` lr or the problem of `scheduler.step()`

About the envirioment

- python=3.6.9
- pytorch=1.1.0

In addition, I can't find this problem when pytorch=0.4.1 is used.

cc @vincentqb

**Comment by vincentqb** [comment]
Please update to the latest version of PyTorch.

As discussed in #26423, `get_lr` was not meant to be the appropriate way of getting the learning rate. A warning is raised in #31125. 

The correct way is `scheduler.get_last_lr` or `optimizer.param_groups[0]["lr"]`.

```
optimizer = optim.SGD(net.parameters(), lr=0.1)
scheduler = optim.lr_scheduler.StepLR(optimizer, step_size=5, gamma=0.5)

for i in range(15):
    lr = scheduler.get_last_lr()[0]
    lr1 = optimizer.param_groups[0]["lr"]
    print(i, lr, lr1)
    scheduler.step()
```

I'll close this PR, but please feel free to re-open if the problem persists.
